lib/superset/guest_token.rb

```diff
@@ -20,13 +22,17 @@ def params
             "id": embedded_dashboard_id.to_s,
             "type": "dashboard" }
         ],
-        "rls": [],
+        "rls": rls_clause, # Ex: [{ "clause": "publisher = 'Nintendo'" }]
         "user": current_user_params
```

Comment:
@soundarya-mv  .. I just had another thought on this.

I was thinking that we could make this `GuestToken` class more flexible and provide a splat of additional params to then add into the Token that is created.

Currently for every Embedded dashboard call we have added the URL Parms `current_user_id: currentUserId,`
as part of [constructing each superset dashboard URL](https://github.com/rdytech/neptune/blob/develop/app/javascript/controllers/superset_load_controller.js#L73C13-L75C15). ... soon to be renamed to `embedded_app_current_user_id: currentUserId`
That would be used in superset with `url_param('current_user_id')

Those url_params could be replaced by having extra params to the Guest Token as well that way any additional params could be included.

For our purposes it could be used as a middle step in the testing process before turning on the RLS for a Dashboard in JobReady.

```
 def initialize(embedded_dashboard_id: , rls_clause: [], **additional_params)

 def params
      {
        "resources": [
          {
            "id": embedded_dashboard_id.to_s,
            "type": "dashboard" }
        ],
        "rls": rls_clause, # Ex: [{ "clause": "publisher = 'Nintendo'" }]
        }.merge(additional_params)
  end
```

Then in JobReady ... when we [construct the Guest Token](https://github.com/rdytech/neptune/pull/10342/files#diff-43ff9b686fdb8b75cc28b847b65686b0b9c0cdc45ddd052c33dfe5df85cd59e3R12)
we would have more flexibility in what params we can add to the Guest Token

```
    guest_token = Superset::GuestToken.new(
      embedded_dashboard_id: dashboard_ss_id,
      rls_clause: RlsFilterClauseBuilder.new(dashboard_ss_id, current_user).build,
      embedded_app_current_user_id: currentUserId,
      embedded_app_user_programmes: currentUserProgrammes,
      embedded_app_user_accessible_sites: currentUserAccessibleSites,
      ).guest_token
```

Those extra params would then be used in Superset as
```'{{ embedded_app_current_user_id }}' `